<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>702</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>高级位域计算器</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QGridLayout" name="layoutTopTable">
      <item row="0" column="0">
       <widget class="QLabel" name="labelExpr">
        <property name="text">
         <string>表达式</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="editExpression"/>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="labelHex">
        <property name="text">
         <string>HEX</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="editHex"/>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="labelDec">
        <property name="text">
         <string>DEC</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLineEdit" name="editDec"/>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="labelOct">
        <property name="text">
         <string>OCT</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLineEdit" name="editOct"/>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="labelBin">
        <property name="text">
         <string>BIN</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QLineEdit" name="editBin"/>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="labelSplit">
        <property name="text">
         <string>BIN分割(an-a0)</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QLineEdit" name="editSplitRule">
        <property name="placeholderText">
         <string>例: 1,2,4 (从高位应用)</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="labelBinRes">
        <property name="text">
         <string>BIN结果(an-a0)</string>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QLineEdit" name="editBinResult"/>
      </item>
      <item row="7" column="0">
       <widget class="QLabel" name="labelDecRes">
        <property name="text">
         <string>DEC结果(an-a0)</string>
        </property>
       </widget>
      </item>
      <item row="7" column="1">
       <widget class="QLineEdit" name="editDecResult"/>
      </item>
      <item row="8" column="0">
       <widget class="QLabel" name="labelHexRes">
        <property name="text">
         <string>HEX结果(an-a0)</string>
        </property>
       </widget>
      </item>
      <item row="8" column="1">
       <widget class="QLineEdit" name="editHexResult"/>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="layoutUpdateMode">
      <item>
       <widget class="QCheckBox" name="chkSyncExpression">
        <property name="text">
         <string>同步表达式（勾选时同步表达式，不勾选仅更新数值）</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QGridLayout" name="gridButtons">
      <item row="6" column="3">
       <widget class="QPushButton" name="btnD">
        <property name="text">
         <string>D</string>
        </property>
       </widget>
      </item>
      <item row="5" column="4">
       <widget class="QPushButton" name="btnEq">
        <property name="text">
         <string>=</string>
        </property>
       </widget>
      </item>
      <item row="3" column="2">
       <widget class="QPushButton" name="btn3">
        <property name="text">
         <string>3</string>
        </property>
       </widget>
      </item>
      <item row="3" column="3">
       <widget class="QPushButton" name="btnSub">
        <property name="text">
         <string>-</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QPushButton" name="btn4">
        <property name="text">
         <string>4</string>
        </property>
       </widget>
      </item>
      <item row="6" column="4">
       <widget class="QPushButton" name="btnE">
        <property name="text">
         <string>E</string>
        </property>
       </widget>
      </item>
      <item row="8" column="2">
       <widget class="QPushButton" name="btnXor">
        <property name="text">
         <string>^</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QPushButton" name="btnA">
        <property name="text">
         <string>A</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QPushButton" name="btn9">
        <property name="text">
         <string>9</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QPushButton" name="btn2">
        <property name="text">
         <string>2</string>
        </property>
       </widget>
      </item>
      <item row="6" column="5">
       <widget class="QPushButton" name="btnF">
        <property name="text">
         <string>F</string>
        </property>
       </widget>
      </item>
      <item row="8" column="1">
       <widget class="QPushButton" name="btnOr">
        <property name="text">
         <string>|</string>
        </property>
       </widget>
      </item>
      <item row="8" column="3">
       <widget class="QPushButton" name="btnNot">
        <property name="text">
         <string>~</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QPushButton" name="btn7">
        <property name="text">
         <string>7</string>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QPushButton" name="btn6">
        <property name="text">
         <string>6</string>
        </property>
       </widget>
      </item>
      <item row="2" column="3">
       <widget class="QPushButton" name="btnMul">
        <property name="text">
         <string>*</string>
        </property>
       </widget>
      </item>
      <item row="3" column="4">
       <widget class="QPushButton" name="btnAdd">
        <property name="text">
         <string>+</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QPushButton" name="btn8">
        <property name="text">
         <string>8</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QPushButton" name="btn5">
        <property name="text">
         <string>5</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QPushButton" name="btn1">
        <property name="text">
         <string>1</string>
        </property>
       </widget>
      </item>
      <item row="0" column="3">
       <widget class="QPushButton" name="btnBackspace">
        <property name="text">
         <string>Back</string>
        </property>
       </widget>
      </item>
      <item row="0" column="4">
       <widget class="QPushButton" name="btnClear">
        <property name="text">
         <string>归零</string>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QPushButton" name="btnB">
        <property name="text">
         <string>B</string>
        </property>
       </widget>
      </item>
      <item row="6" column="2">
       <widget class="QPushButton" name="btnC">
        <property name="text">
         <string>C</string>
        </property>
       </widget>
      </item>
      <item row="8" column="0">
       <widget class="QPushButton" name="btnAnd">
        <property name="text">
         <string>&amp;&amp;</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QPushButton" name="btn0">
        <property name="text">
         <string>0</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QPushButton" name="btnLParen">
        <property name="text">
         <string>(</string>
        </property>
       </widget>
      </item>
      <item row="5" column="2">
       <widget class="QPushButton" name="btnRParen">
        <property name="text">
         <string>)</string>
        </property>
       </widget>
      </item>
      <item row="3" column="5">
       <widget class="QPushButton" name="btnShl">
        <property name="text">
         <string>&lt;&lt;</string>
        </property>
       </widget>
      </item>
      <item row="5" column="5">
       <widget class="QPushButton" name="btnShr">
        <property name="text">
         <string>&gt;&gt;</string>
        </property>
       </widget>
      </item>
      <item row="5" column="3">
       <widget class="QPushButton" name="btnMod">
        <property name="text">
         <string>%</string>
        </property>
       </widget>
      </item>
      <item row="2" column="4">
       <widget class="QPushButton" name="btnDiv">
        <property name="text">
         <string>/</string>
        </property>
       </widget>
      </item>
      <item row="8" column="4">
       <widget class="QPushButton" name="btnComma">
        <property name="text">
         <string>,</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
